[FIX][6.4.0][main] Examples/tests: Use and test `amd_comgr_action_info_set_option_list`

domcharrier opened the pull request:
Routine `amd_comgr_action_info_set_options` will get removed in ROCm 6.4.0. It has been deprecated for a long time now (at least since ROCm 6.1.0). Therefore, this commit uses `amd_comgr_action_info_set_option_list` instead via the 
new interfaces provided by `amd_comgr_pyext.py`.

## TODOs

* [ ] Test via CI job.

domcharrier committed:
examples/test: use+test *action_info_set_option_list

Routine `amd_comgr_action_info_set_options` will get removed in ROCm 6.4.0.
It has been deprecated for a long time now (at least since ROCm 6.1.0). Therefore, this
commit uses `amd_comgr_action_info_set_option_list` instead.

Requires changes to `codegen/base` of ROCm LLVM Python.
Possibly changes need to be backported to other ROCm releases
that are still active.

psamfass reviewed (approved):
LGTM, RTM conditional on passing test!

domcharrier committed:
_rend*_upd*_vers*.py:  git {branch --show-current->rev-parse ...}

Use `git rev-parse --abbrev-ref HEAD` instead of `git branch --show-current` as it is supported by older Git versions too.

domcharrier commented:
Superseded. New releases bundle code generation base branch code, docs, examples and generated bindings.
